File: csharp/DesiArchitect.DesignPatterns/04_Decorator/After/PlatformFeeDecorator.cs
namespace DesiArchitect.DesignPatterns.Decorator.After;

public class PlatformFeeDecorator : IBill
{
    private readonly IBill _inner;
    private readonly decimal _fee = 5.0m;

    public PlatformFeeDecorator(IBill inner) => _inner = inner;

    public decimal Total()
    {
        var subtotal = _inner.Total();
        var result = subtotal + _fee;
        Console.WriteLine($"  🏷️  + Platform fee: ₹{_fee} -> ₹{result}");
        return result;
    }
}

File: typescript/src/04-decorator/after/platform-fee-decorator.ts
import type { IBill } from './ibill.js';

export class PlatformFeeDecorator implements IBill {
  private readonly fee = 5;

  constructor(private readonly inner: IBill) {}

  total(): number {
    const subtotal = this.inner.total();
    const result = subtotal + this.fee;
    console.log(`  🏷️  + Platform fee: ₹${this.fee} -> ₹${result}`);
    return result;
  }
}